parachain/pallets/verifier-lightclient/src/lib.rs

```diff
@@ -469,64 +465,18 @@ fn ancestry<T: Config>(mut hash: H256) -> impl Iterator<Item = (H256, EthereumHe
 	})
 }
 
-impl<T: Config> Verifier<T::AccountId> for Module<T> {
+impl<T: Config> Verifier for Module<T> {
 
-	fn verify(_: T::AccountId, _: AppId, message: &Message) -> DispatchResult {
-		match message.verification {
-			VerificationInput::ReceiptProof { ref block_hash, ref tx_index, ref proof } => {
-				let receipt = Self::verify_receipt_inclusion(block_hash, &proof.1)?;
+	fn verify(message: &Message) -> Result<Log, DispatchError> {
+		let receipt = Self::verify_receipt_inclusion(&message.proof)?;
 
-				let log: Log = rlp::decode(&message.payload)
-					.map_err(|_| Error::<T>::InvalidProof)?;
-				if !receipt.contains_log(&log) {
-					return Err(Error::<T>::InvalidProof.into());
-				}
-		
-				Ok(())
-			},
-			_ => Err(Error::<T>::UnsupportedVerificationScheme.into())
-		}
-	}
-
-	fn verify_bulk(_: T::AccountId, messages: &[(AppId, Message)]) -> DispatchResult {
```

Comment:
Will you add this back in in future? I do think it's worth it, along with `submit_bulk`, mostly for catching up the node after downtime. In another PR though

Reply:
I'll definitely add it back. I originally removed it as the interface some of the implementation would have needed to be updated for the changes in the bridge design, and well I just ran out of time to do that before the audit started.

I'm thinking that I'll change `verify_bulk` to return an iterator that produces values of the form `Result<Log, DispatchError>`. Then the bridge can sequentially iterate over the values and process them or act on the first error found.

Reply:
Added https://github.com/Snowfork/polkadot-ethereum/issues/240 for this.